scripts/build-setup.sh

```diff
@@ -111,22 +111,13 @@ fi
 
 # setup and install conda environment
 if run_step "1"; then
-    # note: lock file must end in .conda-lock.yml - see https://github.com/conda-incubator/conda-lock/issues/154
-    CONDA_REQS=$CYDIR/conda-reqs
-    CONDA_LOCK_REQS=$CONDA_REQS/conda-lock-reqs
-    # must match with the file generated by generate-conda-lockfile.sh
-    LOCKFILE=$CONDA_LOCK_REQS/conda-requirements-$TOOLCHAIN_TYPE-linux-64.conda-lock.yml
-
-    if [ "$USE_UNPINNED_DEPS" = true ]; then
-        # auto-gen the lockfiles
-        $CYDIR/scripts/generate-conda-lockfiles.sh
-    fi
-
-    # use conda-lock to create env
-    conda-lock install --conda $(which conda) -p $CYDIR/.conda-env $LOCKFILE
-
-    source $CYDIR/.conda-env/etc/profile.d/conda.sh
-    conda activate $CYDIR/.conda-env
+    try; (
+        source $CYDIR/scripts/build-step-init-conda-environment.sh
+    )
+    catch || {
+        echo "Build script exited with exit code $? at step 1: conda environment setup. Check the above logs for more details on the error."
```

Comment:
Nit: Since we are abstracting things out even more, can we put this in a function as well?